crates/common/consensus/src/predicates.rs

```diff
@@ -1,7 +1,63 @@
-use crate::attestation_data::AttestationData;
+use crate::{
+    attestation_data::AttestationData, deneb::beacon_state::BeaconState,
+    fork_choice::helpers::constants::DOMAIN_BEACON_ATTESTER,
+    indexed_attestation::IndexedAttestation, misc::compute_signing_root,
+};
 
 /// Check if ``data_1`` and ``data_2`` are slashable according to Casper FFG rules.
-pub fn is_slashable_attestation_data(data_1: AttestationData, data_2: AttestationData) -> bool {
+pub fn is_slashable_attestation_data(data_1: &AttestationData, data_2: &AttestationData) -> bool {
     (data_1 != data_2 && data_1.target.epoch == data_2.target.epoch)
         || (data_1.source.epoch < data_2.source.epoch && data_2.target.epoch < data_1.target.epoch)
 }
+
+pub fn is_valid_indexed_attestation(
+    state: &BeaconState,
+    indexed_attestation: &IndexedAttestation,
+) -> bool {
+    let indices: Vec<usize> = indexed_attestation
+        .attesting_indices
+        .iter()
+        .map(|&i| i as usize)
+        .collect();
+    // Verify indices are sorted and unique
+    if indices.is_empty() || !is_sorted_and_unique(&indices) {
+        return false;
+    }
+
+    // Collect public keys of attesting validators
+    let pubkeys: Vec<_> = indices
+        .iter()
+        .filter_map(|&i| state.validators.get(i).map(|v| v.pubkey.clone()))
+        .collect();
+
+    // Compute domain and signing root
+    let domain = match state.get_domain(
```

Comment:
get_domain doesn't need to return a result, if you want to remove it in this PR I am cool with that, or I can, I think it is a little awkward to return false here